pkg/controller/pod.go

```diff
@@ -588,8 +599,10 @@ func (c *Controller) reconcileAllocateSubnets(pod *v1.Pod, needAllocatePodNets [
 				}
 			}
 
-			if securityGroupAnnotation != "" {
-				sgNames := strings.Split(securityGroups, ",")
+			if securityGroupAnnotation != "" || oldSgList != nil {
+				securityGroups := strings.ReplaceAll(securityGroupAnnotation, " ", "")
+				newSgList := strings.Split(securityGroups, ",")
+				sgNames := util.UnionStringSlice(oldSgList, newSgList)
```

Comment:
What happens if all security groups are deleted before starting the stopped VM?

Reply:
sg not exist, sg update logic will not enque

Reply:
My fault. OR logic is used so security groups will be updated as expected.